Source/Utils/WarpXAlgorithmSelection.cpp

```diff
@@ -0,0 +1,88 @@
+#include <WarpXAlgorithmSelection.H>
+
+// Define dictionary with correspondance between user-input strings,
+// and corresponding integer for use inside the code (e.g. in PICSAR).
+
+const std::map<std::string, int> maxwell_solver_algo_to_int = {
+    {"yee",     MaxwellSolverAlgo::Yee },
+#ifndef AMREX_USE_GPU // Only available on CPU
+    {"ckc",     MaxwellSolverAlgo::CKC },
+#endif
+    {"default", MaxwellSolverAlgo::Yee }
+};
+
+const std::map<std::string, int> particle_pusher_algo_to_int = {
+    {"boris",   ParticlePusherAlgo::Boris },
+    {"vay",     ParticlePusherAlgo::Vay },
+    {"default", ParticlePusherAlgo::Boris }
+};
+
+const std::map<std::string, int> current_deposition_algo_to_int = {
+    {"esirkepov",            CurrentDepositionAlgo::Esirkepov },
+    {"direct",               CurrentDepositionAlgo::Direct },
+#if (!defined AMREX_USE_GPU)&&(AMREX_SPACEDIM == 3) // Only available on CPU and 3D
+    {"direct-vectorized",    CurrentDepositionAlgo::DirectVectorized },
+#endif
+    {"default",              CurrentDepositionAlgo::Esirkepov }
+};
+
+const std::map<std::string, int> charge_deposition_algo_to_int = {
+    {"standard",   ChargeDepositionAlgo::Standard },
+#if (!defined AMREX_USE_GPU)&&(AMREX_SPACEDIM == 3) // Only available on CPU and 3D
+    {"vectorized", ChargeDepositionAlgo::Vectorized },
+    {"default",    ChargeDepositionAlgo::Vectorized }
+#else
+    {"default",    ChargeDepositionAlgo::Standard }
+#endif
+};
+
+const std::map<std::string, int> gathering_algo_to_int = {
+    {"standard",   GatheringAlgo::Standard },
+#ifndef AMREX_USE_GPU // Only available on CPU
+    {"vectorized", GatheringAlgo::Vectorized },
+    {"default",    GatheringAlgo::Vectorized }
+#else
+    {"default",    GatheringAlgo::Standard }
+#endif
+};
+
+
+int
+GetAlgorithmInteger( amrex::ParmParse& pp, const char* pp_search_key ){
+
+    // Read user input ; use "default" if it is not found
+    std::string algo = "default";
+    pp.query( pp_search_key, algo );
+
```

Comment:
You could use `std::transform(algo.begin(), algo.end(), algo.begin(), ::tolower);` here so that users could put either `Boris` or 'boris`, say.